File: sample/src/main/java/com/pagecall/sample/MainActivity.java
package com.pagecall.sample;

import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.WindowManager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
        // handle file upload
        super.onActivityResult(requestCode, resultCode, intent);
        PagecallFragment pagecallFragment = getPagecallFragment();
        if (pagecallFragment != null) {
            pagecallFragment.processActivityResult(requestCode, resultCode, intent);
        }
    }

    private PagecallFragment getPagecallFragment() {
        Fragment navHostFragment = getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment_content_main);
        if (navHostFragment != null) {
            Fragment currentFragment = navHostFragment.getChildFragmentManager().getFragments().get(0);
            if (currentFragment instanceof PagecallFragment) {
                return ((PagecallFragment) currentFragment);
            }
        }
        return null;
    }
}

File: sample_x200/src/main/java/com/example/sample_x200/MainActivity.kt
package com.example.sample_x200

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Button
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    var mainWebView: WebView? = null
    private val startForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
        if (result.resultCode == Activity.RESULT_OK) {
            // 결과가 OK일 때 처리할 로직
            val intent = result.data
            // 인텐트에서 데이터 처리
        }
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mainWebView = findViewById<WebView>(R.id.webView)
        val customHtml = """
            <html>
                <body>
                    <h1>Hello World!</h1>
                    <a href="https://app.pagecall.com/meet?room_id=6572e3bc435814217603baa1">Go to pagecall room</a>
                </body>
            </html>""".trimIndent()
        mainWebView?.loadData(customHtml, "text/html", "UTF-8")
        mainWebView?.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                if (Uri.parse(url).host == "app.pagecall.com") {
                    Log.d("Ryan123", "Trying to go to pagecall room")
                    val intent = Intent(this@MainActivity, PagecallActivity::class.java).apply {
                        putExtra("URL", url)
                    }
                    startForResult.launch(intent)
                    return true // WebView가 URL을 로드하지 않도록 함
                }
                return false // 다른 URL은 WebView가 로드하도록 함
            }
        }
    }
}
